src/napari_spatialdata/_view.py

```diff
@@ -19,12 +19,203 @@
+        self.x_selection_widget = QComboBox()
+        self.x_selection_widget.addItem("obsm", None)
+        self.x_selection_widget.addItem("obs", None)
+        self.x_selection_widget.addItem("var", None)
+
+        self.layout().addWidget(x_selection_label)
+        self.layout().addWidget(self.x_selection_widget)
+
+        # X-axis
+        x_label = QLabel("Select x-axis:")
+        x_label.setToolTip("Select layer to visualise in x-axis.")
+
+        self.x_widget = ScatterListWidget(self.viewer, self.model, attr="obsm")
+        self.x_widget.setAttribute("obsm")
+
+        self.x_component_widget = ComponentWidget(self.model, attr="obsm")
+        self.x_component_widget.setToolTip("obsm")
+        self.x_component_widget.currentTextChanged.connect(self.x_widget.setComponent)
+        self.x_widget.itemClicked.connect(self.x_component_widget._onClickChange)
+
+        self.layout().addWidget(x_label)
+        self.layout().addWidget(self.x_widget)
+        self.layout().addWidget(self.x_component_widget)
+
+        self.x_selection_widget.currentTextChanged.connect(self.x_widget.setAttribute)
+        self.x_selection_widget.currentTextChanged.connect(self.x_component_widget.setAttribute)
+        self.x_selection_widget.currentTextChanged.connect(self.x_component_widget.setToolTip)
+
+        # Y selection
+        y_selection_label = QLabel("Select type for Y axis:")
+        y_selection_label.setToolTip("Select between obs, obsm and var.")
+
+        self.y_selection_widget = QComboBox()
+        self.y_selection_widget.addItem("obsm", None)
+        self.y_selection_widget.addItem("obs", None)
+        self.y_selection_widget.addItem("var", None)
+
+        self.layout().addWidget(y_selection_label)
+        self.layout().addWidget(self.y_selection_widget)
+
+        # Y-axis
+        y_label = QLabel("Select y-axis:")
+        y_label.setToolTip("Select layer to visualise in y-axis.")
+        self.y_widget = ScatterListWidget(self.viewer, self.model, attr="obsm")
+        self.y_widget.setAttribute("obsm")
+
+        self.y_component_widget = ComponentWidget(self.model, attr="obsm")
+        self.y_component_widget.setToolTip("obsm")
+        self.y_component_widget.currentTextChanged.connect(self.y_widget.setComponent)
+        self.y_widget.itemClicked.connect(self.y_component_widget._onClickChange)
+
+        self.layout().addWidget(y_label)
+        self.layout().addWidget(self.y_widget)
+        self.layout().addWidget(self.y_component_widget)
+
+        self.y_selection_widget.currentTextChanged.connect(self.y_widget.setAttribute)
+        self.y_selection_widget.currentTextChanged.connect(self.y_component_widget.setAttribute)
+        self.y_selection_widget.currentTextChanged.connect(self.y_component_widget.setToolTip)
+
+        # Color
+        color_selection_label = QLabel("Select type for color:")
+        color_selection_label.setToolTip("Select between obs and var.")
```

Comment:
there is a lot of code repetition here. Is there maybe a way to put this in a widget itself? like https://github.com/scverse/napari-spatialdata/blob/fccc67a08b5f5073276e9f9fd03937daa6d8e843/src/napari_spatialdata/_widgets.py#L302